From 664e08aa6f8d989fd9ca4c1c7a2109736f0fd5ac Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <david.abdurachmanov@gmail.com>
Date: Fri, 8 Oct 2021 11:30:49 +0000
Subject: [PATCH 1/9] riscv: SiFive Unleashed booti compressed kernel support

Signed-off-by: David Abdurachmanov <david.abdurachmanov@gmail.com>
---
 include/configs/sifive-unleashed.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/sifive-unleashed.h b/include/configs/sifive-unleashed.h
index b6c29f8c..db0ebaf9 100644
--- a/include/configs/sifive-unleashed.h
+++ b/include/configs/sifive-unleashed.h
@@ -71,6 +71,8 @@
 	"script_size_f=0x1000\0" \
 	"pxefile_addr_r=0x88200000\0" \
 	"ramdisk_addr_r=0x88300000\0" \
+	"kernel_comp_addr_r=0x90300000\0" \
+	"kernel_comp_size=0x4000000\0" \
 	"type_guid_gpt_loader1=" TYPE_GUID_LOADER1 "\0" \
 	"type_guid_gpt_loader2=" TYPE_GUID_LOADER2 "\0" \
 	"type_guid_gpt_system=" TYPE_GUID_SYSTEM "\0" \
-- 
2.32.0

