From 2ee3bdd54d7dbfabf512a5496ad9794be45849f2 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Sun, 1 Sep 2013 21:42:48 -0500
Subject: [PATCH 02/16] add option to include generic distro config

---
 include/common.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/common.h b/include/common.h
index 8addf43..113df4a 100644
--- a/include/common.h
+++ b/include/common.h
@@ -99,6 +99,11 @@ typedef volatile unsigned char	vu_char;
 #include <flash.h>
 #include <image.h>
 
+/* use generic distro config */
+#ifdef DISTRO_DEFAULTS
+#include <config_distro_default.h>
+#endif
+
 #ifdef DEBUG
 #define _DEBUG	1
 #else
-- 
1.8.3.1

