From ad30cfb1284981c5c3f1d3bb92e1ef417546947e Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Mon, 2 Sep 2013 11:26:39 -0500
Subject: [PATCH 03/16] set omap4 boards to use the generic distro support

---
 include/configs/omap4_common.h | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/include/configs/omap4_common.h b/include/configs/omap4_common.h
index e9f2383..634668a 100644
--- a/include/configs/omap4_common.h
+++ b/include/configs/omap4_common.h
@@ -106,6 +106,9 @@
 /* commands to include */
 #include <config_cmd_default.h>
 
+/* enable generic distro config */
+#define DISTRO_DEFAULTS 1
+
 /* Enabled commands */
 #define CONFIG_CMD_EXT2		/* EXT2 Support                 */
 #define CONFIG_CMD_FAT		/* FAT support                  */
@@ -132,9 +135,13 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"loadaddr=0x82000000\0" \
+	"pxefile_addr_r=0x82000000\0" \
+	"kernel_addr_r=0x84000000\0" \
+	"ramdisk_addr_r=0x90000000\0" \
+	"fdt_addr_r=0x83000000\0" \
 	"console=ttyO2,115200n8\0" \
 	"fdt_high=0xffffffff\0" \
-	"fdtaddr=0x80f80000\0" \
+	"fdt_addr=0x80f80000\0" \
 	"fdtfile=undefined\0" \
 	"bootpart=0:2\0" \
 	"bootdir=/boot\0" \
-- 
1.8.3.1

