From 4fe19544ffda98c8ab9eea55d29f8c9179716e88 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Mon, 2 Sep 2013 11:33:46 -0500
Subject: [PATCH 04/15] set wandboard to use generic commands and set needed
 variables

---
 include/configs/wandboard.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/configs/wandboard.h b/include/configs/wandboard.h
index 442a984..423d94e 100644
--- a/include/configs/wandboard.h
+++ b/include/configs/wandboard.h
@@ -40,6 +40,9 @@
 #define CONFIG_CONS_INDEX		1
 #define CONFIG_BAUDRATE			115200
 
+/* enable generic distro config */
+#define DISTRO_DEFAULTS 1
+
 /* Command definition */
 #include <config_cmd_default.h>
 
@@ -114,6 +117,10 @@
 	"initrd_high=0xffffffff\0" \
 	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"fdt_addr=0x11000000\0" \
+	"pxefile_addr_r=0x13000000\0" \
+	"kernel_addr_r=0x14000000\0" \
+	"ramdisk_addr_r=0x32000000\0" \
+	"fdt_addr_r=0x12000000\0" \
 	"boot_fdt=try\0" \
 	"ip_dyn=yes\0" \
 	"mmcdev=" __stringify(CONFIG_SYS_MMC_ENV_DEV) "\0" \
-- 
1.8.3.1

