From 0a50f653c44cef7685d8300ddccb2c7476cbed71 Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Wed, 15 Jul 2015 12:37:44 +0100
Subject: [PATCH 6/9] mx6: utilite: disable logos

---
 include/configs/cm_fx6.h | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/include/configs/cm_fx6.h b/include/configs/cm_fx6.h
index 8a2e46c..713d90a 100644
--- a/include/configs/cm_fx6.h
+++ b/include/configs/cm_fx6.h
@@ -212,12 +212,4 @@
 #define CONFIG_CONSOLE_MUX
 #define CONFIG_VIDEO_SW_CURSOR
 
-#define CONFIG_SPLASH_SCREEN
-#define CONFIG_SPLASH_SOURCE
-#define CONFIG_CMD_BMP
-#define CONFIG_VIDEO_BMP_RLE8
-
-#define CONFIG_VIDEO_LOGO
-#define CONFIG_VIDEO_BMP_LOGO
-
 #endif	/* __CONFIG_CM_FX6_H */
-- 
2.5.0

