From 0f680a1e121c91be46ed1c159102536f50ee7448 Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <david.abdurachmanov@gmail.com>
Date: Fri, 8 Oct 2021 11:52:13 +0000
Subject: [PATCH 7/9] riscv: add compressed kernel support for qemu-riscv

Signed-off-by: David Abdurachmanov <david.abdurachmanov@gmail.com>
---
 include/configs/qemu-riscv.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/qemu-riscv.h b/include/configs/qemu-riscv.h
index bbeea96e..0ede2a51 100644
--- a/include/configs/qemu-riscv.h
+++ b/include/configs/qemu-riscv.h
@@ -64,6 +64,8 @@
 	"scriptaddr=0x88100000\0" \
 	"pxefile_addr_r=0x88200000\0" \
 	"ramdisk_addr_r=0x88300000\0" \
+	"kernel_comp_addr_r=0x90300000\0" \
+	"kernel_comp_size=0x4000000\0" \
 	BOOTENV
 #endif
 
-- 
2.32.0

