From c1e6848661b1ceb9e912a9408da4a00cdd11e85f Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Mon, 2 Sep 2013 15:27:53 -0500
Subject: [PATCH 07/16] enable CONFIG_CMD_BOOTMENU for distro configs

---
 include/config_distro_default.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/config_distro_default.h b/include/config_distro_default.h
index b892868..99bb4b1 100644
--- a/include/config_distro_default.h
+++ b/include/config_distro_default.h
@@ -25,6 +25,7 @@
 #define CONFIG_OF_LIBFDT
 
 #define CONFIG_CMD_BOOTZ
+#define CONFIG_CMD_BOOTMENU
 #define CONFIG_CMD_DHCP
 #define CONFIG_CMD_ELF
 #define CONFIG_CMD_EXT2
-- 
1.8.3.1

