From 2394d596e963d5532f3ff00e097ebcf5c8338610 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Thu, 17 Oct 2013 17:55:53 -0500
Subject: [PATCH 08/15] disable FIT image support since it fails to build

---
 include/config_distro_default.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/config_distro_default.h b/include/config_distro_default.h
index aa7500e..56a5557 100644
--- a/include/config_distro_default.h
+++ b/include/config_distro_default.h
@@ -39,7 +39,6 @@
 #define CONFIG_AUTO_COMPLETE
 #define CONFIG_BOOTDELAY     2
 #define CONFIG_SYS_LONGHELP
-#define CONFIG_FIT
 #define CONFIG_MENU
 #define CONFIG_DOS_PARTITION
 #define CONFIG_EFI_PARTITION
-- 
1.8.3.1

