From bd6c9bae1941c74cb9c963a702ff491a57d61ab3 Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Fri, 6 Mar 2015 10:49:53 +0000
Subject: [PATCH 09/14] RiOT board: set console speed

---
 include/configs/embestmx6boards.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/embestmx6boards.h b/include/configs/embestmx6boards.h
index b4b3ae8..b8696a3 100644
--- a/include/configs/embestmx6boards.h
+++ b/include/configs/embestmx6boards.h
@@ -22,7 +22,7 @@
 #define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_MXC_UART_BASE		UART2_BASE
-#define CONFIG_CONSOLE_DEV		"ttymxc1"
+#define CONFIG_CONSOLE_DEV		"ttymxc1,115200"
 #define CONFIG_MMCROOT			"/dev/mmcblk1p2"
 
 #define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
-- 
2.3.2

