From d0bf7b6dd3ebe7d9bbf2a84cbb3e8d80f576115c Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Mon, 2 Sep 2013 18:26:55 -0500
Subject: [PATCH 10/16] remove USB from distro default, not all systems support
 USB

---
 include/config_distro_default.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/config_distro_default.h b/include/config_distro_default.h
index 8c786d2..65a0c17 100644
--- a/include/config_distro_default.h
+++ b/include/config_distro_default.h
@@ -38,7 +38,6 @@
 #define CONFIG_CMD_NET
 #define CONFIG_CMD_PING
 #define CONFIG_CMD_PXE
-#define CONFIG_CMD_USB
 
 #define CONFIG_CMDLINE_EDITING
 #define CONFIG_AUTO_COMPLETE
@@ -50,6 +49,5 @@
 #define CONFIG_EFI_PARTITION
 #define CONFIG_SUPPORT_RAW_INITRD
 #define CONFIG_SYS_HUSH_PARSER
-#define CONFIG_USB_STORAGE
 
 #endif	/* _CONFIG_CMD_DISTRO_DEFAULT_H */
-- 
1.8.3.1

