From fcadba88c2e9d7214b25a8ed4d263ebbc22f0680 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Mon, 21 Apr 2014 12:31:22 -0500
Subject: [PATCH 12/14] cleanup duplicate options in paz00 config

---
 include/configs/paz00.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/configs/paz00.h b/include/configs/paz00.h
index dd0abf8..8fc2b5a 100644
--- a/include/configs/paz00.h
+++ b/include/configs/paz00.h
@@ -61,8 +61,6 @@
 #define CONFIG_USB_ETHER_ASIX
 
 /* General networking support */
-#define CONFIG_CMD_NET
-#define CONFIG_CMD_DHCP
 
 /* LCD support */
 #define CONFIG_LCD
-- 
1.9.0

