From 3f7058a106fde4bf69b76933a1e2f56a051d8466 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Tue, 18 Mar 2014 16:13:38 -0500
Subject: [PATCH 12/13] check for fdtfile and fdt_file to find the devicetree
 file

---
 common/cmd_pxe.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/common/cmd_pxe.c b/common/cmd_pxe.c
index 3483328..c58e471 100644
--- a/common/cmd_pxe.c
+++ b/common/cmd_pxe.c
@@ -712,6 +712,13 @@ static int label_boot(cmd_tbl_t *cmdtp, struct pxe_label *label)
 			char *f1, *f2, *f3, *f4, *slash;
 
 			f1 = getenv("fdtfile");
+			/*
+			 * some boards have used fdt_file as the environment variable for
+			 * defining the device tree file so lets check for fdt_file also.
+			 */
+			if (!f1) {
+				f1 = getenv("fdt_file");
+			}
 			if (f1) {
 				f2 = "";
 				f3 = "";
-- 
1.9.0

