From f5c7ad35415218c20b904edf1981a33a47bcdf92 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Thu, 17 Oct 2013 18:55:25 -0500
Subject: [PATCH 14/16] Use SPDX header in distro config

---
 include/config_distro_default.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/include/config_distro_default.h b/include/config_distro_default.h
index 132da4f..bd9b4a0 100644
--- a/include/config_distro_default.h
+++ b/include/config_distro_default.h
@@ -1,9 +1,7 @@
 /*
  * Copyright 2013 Red Hat, Inc.
  *
- * This file is licensed under the terms of the GNU General Public
- * License Version 2. This file is licensed "as is" without any
- * warranty of any kind, whether express or implied.
+ * SPDX-License-Identifier:     GPL-2.0+
  */
 
 #ifndef _CONFIG_CMD_DISTRO_DEFAULT_H
-- 
1.8.3.1

