From 8ef41c6750f256383ea38ee7aa6d03924cdeeeb4 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Mon, 21 Apr 2014 14:44:07 -0500
Subject: [PATCH 15/36] omap4 buildfixes

---
 include/configs/ti_omap4_common.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/include/configs/ti_omap4_common.h b/include/configs/ti_omap4_common.h
index 88b0264..a1ebfb5 100644
--- a/include/configs/ti_omap4_common.h
+++ b/include/configs/ti_omap4_common.h
@@ -88,10 +88,6 @@
 #define CONFIG_USB_TTY			1
 #define CONFIG_SYS_CONSOLE_IS_IN_ENV	1
 
-/* Per-Soc commands */
-#undef CONFIG_CMD_NET
-#undef CONFIG_CMD_NFS
-
 /*
  * Environment setup
  */
-- 
1.9.0

