--- u-boot-2020.10/common/spl/Kconfig.orig	2020-10-07 12:08:54.972632715 +0100
+++ u-boot-2020.10/common/spl/Kconfig	2020-10-07 12:08:59.100672506 +0100
@@ -324,7 +324,7 @@
 	default 0x200 if ARCH_SOCFPGA || ARCH_AT91
 	default 0x300 if ARCH_ZYNQ || ARCH_KEYSTONE || OMAP34XX || OMAP44XX || \
 		         OMAP54XX || AM33XX || AM43XX || ARCH_K3
-	default 0x4000 if ARCH_ROCKCHIP
+	default 0x200 if ARCH_ROCKCHIP
 	default 0x822 if TARGET_SIFIVE_FU540
 	help
 	  Address on the MMC to load U-Boot from, when the MMC is being used
