From 961918a6fcbab383c92135f5cad5ea758bdaef79 Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Wed, 4 Dec 2019 23:27:10 +0000
Subject: [PATCH] tools fix

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 556cebdfc4..4bcf4d677f 100644
--- a/Makefile
+++ b/Makefile
@@ -1894,6 +1894,7 @@ checkarmreloc: u-boot
 	fi
 
 tools/version.h: include/version.h
+	$(Q)mkdir -p $(dir $@)
 	$(call if_changed,copy)
 
 envtools: scripts_basic $(version_h) $(timestamp_h) tools/version.h
-- 
2.23.0

