diff --git a/include/configs/qemu-riscv.h b/include/configs/qemu-riscv.h
index bc831a84..442f0f9d 100644
--- a/include/configs/qemu-riscv.h
+++ b/include/configs/qemu-riscv.h
@@ -60,6 +60,8 @@
 	"scriptaddr=0x88100000\0" \
 	"pxefile_addr_r=0x88200000\0" \
 	"ramdisk_addr_r=0x88300000\0" \
+	"kernel_comp_addr_r=0x90000000\0" \
+	"filesize=0x4000000\0" \
 	BOOTENV
 #endif
 
diff --git a/include/configs/sifive-fu540.h b/include/configs/sifive-fu540.h
index 59cf0281..5114dde1 100644
--- a/include/configs/sifive-fu540.h
+++ b/include/configs/sifive-fu540.h
@@ -40,6 +40,8 @@
 	"scriptaddr=0x88100000\0" \
 	"pxefile_addr_r=0x88200000\0" \
 	"ramdisk_addr_r=0x88300000\0" \
+	"kernel_comp_addr_r=0x90000000\0" \
+	"filesize=0x4000000\0" \
 	BOOTENV
 
 #endif /* __CONFIG_H */
